### PR TITLE
removing intentional panics and handles errors gracefully

### PR DESCRIPTION
**Pull request by Streppel**
@JohnRoesler

### CODE REVIEW

File: gocron.go [hunk 1]
```diff
@@ -96,29 +95,27 @@ func Jobs() []*Job {
 }
 
 func formatTime(t string) (hour, min, sec int, err error) {
-	var er = errors.New("time format error")
 	ts := strings.Split(t, ":")
 	if len(ts) < 2 || len(ts) > 3 {
-		err = er
-		return
+		return 0, 0, 0, ErrTimeFormat
```
Comment (Streppel) [inline gocron.go]: this will look leaner on a next PR where we will return a struct instead of bare values

File: job.go [hunk 2]
```diff
@@ -45,46 +53,55 @@ func (j *Job) shouldRun() bool {
...
-		if ok, err := locker.Lock(key); err != nil || !ok {
-			return nil, err
-		}
-
-		defer func() {
-			if e := locker.Unlock(key); e != nil {
-				err = e
-			}
-		}()
+		locker.Lock(key)
```
Comment (Streppel) [inline job.go]: will change our locker signature in the future to remove errors just like Sync.Locker

File: job.go [hunk 3]
```diff
@@ -45,46 +53,55 @@ func (j *Job) shouldRun() bool {
...
+		defer locker.Unlock(key)
 	}
 
 	j.lastRun = time.Now()
-	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
-	j.scheduleNextRun()
-	return
+	if err := j.scheduleNextRun(); err != nil {
```
Comment (Streppel) [inline job.go]: attention here: I swiped execution logic -> scheduleNextRun happens before calljobWithParams now. 

In the earlier implementation, if callJobWithParams takes long to execute, next run schedule will be impacted. I thought we had fixed this in the past (?)

**Comment by JohnRoesler** [comment]
✅ I like it. Just my thought process on this: I was wondering about a sophisticated model for passing errors out. But…the “job” passed by the client can implement whatever error handling it wants. Here we’re only concerned about error handling for gocron specific errors

**Comment by Streppel** [comment]
@JohnRoesler absolutely. A client could easily create an err channel encapsulate through a closure to handle business logic errors or stuff like that. If you're ok with this I'll merge as soon as I can, just running some final tests before that (checking the scheduleNextRun() change specifically)

**Pull request merged**